src/perception/ObjectDatabase.cpp

```diff
@@ -0,0 +1,85 @@
+#include <aikido/perception/ObjectDatabase.hpp>
+
+#include <dart/common/Console.hpp>
+#include <dart/common/LocalResourceRetriever.hpp>
+#include <yaml-cpp/exceptions.h>
+
+namespace aikido {
+namespace perception {
+
+//==============================================================================
+ObjectDatabase::ObjectDatabase(
+    const dart::common::ResourceRetrieverPtr& resourceRetriever,
+    dart::common::Uri configDataURI)
+{
+  // Read the JSON file into string
+  if (!resourceRetriever)
+  {
+    throw std::invalid_argument("ResourceRetrieverPtr given is null!");
+  }
+
+  const dart::common::ResourcePtr resource
```

Comment:
I think this and the following section that puts the file in a string can be replaced by `resourceRetriever->readAll(configDataURI)`.